### DISCUSSION TITLE
Seasonality in lightweight

### DISCUSSION THREAD

**Discussion by Youssef-chaneb** [google/lightweight_mmm#50] [Q&A]
Hello Everyone, 

I want to thank you first for sharing this MMM library.

The model is meeting poor results and after digging, I've found out that the seasonality function modeled by  lightweight is very flat and has not capted the variance of the target.

I've removed the seasonality feature of lightweight in models.py and added a seasonality feature previously calculed with the prophet library as an extra_feature variable.
 
It turns out that the posterior Beta of the seasonality seems to "converge" to 0. I've changed the extra_features prior to a Normal(1,1) distribution and it still gives the same results . 

Find below a graph of the posterior distribution of the Beta seasonality : 

![image](https://user-images.githubusercontent.com/51270670/182123591-67006247-b37e-4cb5-8246-bb513c8c1afb.png)

Do you have any idea why the Beta posterior seems to be centered around 0 with a very low variance ? 

Thank you for your help !

Youssef

**Comment by pabloduque0** [comment]
Hey Youssef! Thanks for opening up this one!

I need to ask a few follow up questions in order to try to understand more about what is going on.

- Can you confirm the model has converged?
- Could you expand a bit more on the model has poor results? On what exactly? and how bad are they?
- Is it national or geo model?
- Did you scale the data? How?
- What other extra features are you using? (If any)
